src/ApparentHorizons/YlmSpherepack.hpp

```diff
@@ -0,0 +1,231 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "YlmSpherepackStorage.hpp"
+
+/// \ingroup Spectral
+/// \brief
+/// C++ interface to SPHEREPACK.
+class YlmSpherepack {
+ public:
+  using first_deriv_t = tnsr::i<DataVector, 2, Frame::Logical>;
+  using second_deriv_t = tnsr::ij<DataVector, 2, Frame::Logical>;
+  using interpolation_info_t = std::vector<std::vector<double>>;
+
+  /// Here l_max and m_max are the largest fully-represented l and m in
+  /// the Ylm expansion.
+  YlmSpherepack(size_t l_max, size_t m_max) noexcept;
+
+  bool operator==(const YlmSpherepack& rhs) const noexcept;
+
+  /// Static functions to return the correct sizes of vectors of
+  /// collocation points and spectral coefficients for a given l_max
+  /// and m_max.  Useful for allocating space without having to create
+  /// a YlmSpherepack.
+  static size_t physical_size(const size_t l_max, const size_t m_max) noexcept {
+    return (l_max + 1) * (2 * m_max + 1);
+  }
+  static size_t spectral_size(const size_t l_max, const size_t m_max) noexcept {
+    return (2 * (l_max + 1) * (m_max + 1));
+  }
+
+  /// Sizes in physical and spectral space for this instance.
+  size_t l_max() const noexcept { return l_max_; }
+  size_t m_max() const noexcept { return m_max_; }
+  size_t physical_size() const noexcept { return n_th_ * n_ph_; }
+  size_t spectral_size() const noexcept { return spectral_extents_[0]; }
+
+  const std::vector<size_t>& physical_extents() const noexcept {
+    return physical_extents_;
+  }
+  /// Spectral extents is 1-dimensional, not 2-dimensional, because
+  /// the basis function cannot be represented as a tensor product.
+  const std::vector<size_t>& spectral_extents() const noexcept {
+    return spectral_extents_;
+  }
+
+  /// Collocation points theta and phi.
+  const std::vector<double>& theta_points() const noexcept;
+  const std::vector<double>& phi_points() const noexcept;
+  std::array<DataVector, 2> theta_phi_points() const noexcept;
+
+  /// Spectral transformations.
```

Comment:
Group comments. Also we should use backticks for variable names, e.g. loop_over_offset -> `loop_over_offset`